src/index/io_flags.rs

```diff
@@ -0,0 +1,24 @@
+//! Module containing the io flags.
+
+/// Io Flags used during index reading, not all flags applicable to all indices
+// This is a set of constants rather than enum so that bitwise operations exist
+pub mod io_flag {
```

Comment:
I was picturing more of a strongly typed bitset type (to prevent the user from making up flag values), but this would do for now.

Reply:
Yeah, I explored this but it seemed a bit heavy weight given the crates I looked at, did you have a specific best practice among them (or alternative) in mind? https://github.com/Enet4/faiss-rs/pull/50#issue-1149999616

Reply:
No additional dependencies would be needed here, just an application of the newtype pattern. Here's the gist:
```rust
#[derive(Debug, Copy, Clone, Eq, Hash, PartialEq)]
pub struct IoFlags(u16);

impl IoFlags {
    /// Load entire index into memory (default behavior)
    pub const MEM_RESIDENT: Self = IoFlags(0x00);
    /// Memory-map index
    pub const MEM_MAP: Self = IoFlags(0x01);
    /// Index is read-only
    pub const READ_ONLY: Self = IoFlags(0x02);
}

impl std::ops::BitOr for IoFlags {
    type IoFlags = Self;

    fn bitor(self, rhs: Self) -> Self::Output {
        Self(self.0 | rhs.0)
    }
}
```

I also increased the width here, it's a bit more future-proof than `u8`.

Reply:
I'm working on taking this suggestion. The gist doesn't work directly yet:

```
error[E0046]: not all trait items implemented, missing: `Output`
  --> src/index/io_flags.rs:17:1
   |
17 | impl std::ops::BitOr for IoFlags {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `Output` in implementation
   |
   = help: implement the missing item: `type Output = Type;`
```

Reply:
Ah I see, need to do `Output = Self` (had to read :)).